File: apps/multimodal-screener/iOS/Aiberry/Emotion/EmotionStyle.swift
import SwiftUI

/// Visual identity (emoji + color) for each emotion label, shared by the voice
/// model, the face model, and the fused readout.
enum EmotionStyle {
    static func emoji(_ label: String) -> String {
        switch label {
        case "Angry":    return "😠"
        case "Disgust":  return "🤢"
        case "Fear":     return "😨"
        case "Happy":    return "😄"
        case "Neutral":  return "😐"
        case "Sad":      return "😢"
        case "Surprise": return "😲"
        default:          return "🫥"
        }
    }

    static func color(_ label: String) -> Color {
        switch label {
        case "Angry":    return Color(red: 0.90, green: 0.30, blue: 0.30)
        case "Disgust":  return Color(red: 0.45, green: 0.65, blue: 0.30)
        case "Fear":     return Color(red: 0.55, green: 0.40, blue: 0.85)
        case "Happy":    return Color(red: 0.97, green: 0.74, blue: 0.20)
        case "Neutral":  return Color(red: 0.55, green: 0.58, blue: 0.62)
        case "Sad":      return Color(red: 0.30, green: 0.55, blue: 0.90)
        case "Surprise": return Color(red: 0.95, green: 0.55, blue: 0.25)
        default:          return Theme.accent
        }
    }
}

/// A single (label, probability) pair, used across voice / face / fused results.
struct EmotionScore: Identifiable, Equatable {
    let id = UUID()
    let label: String
    let probability: Float
}

File: apps/multimodal-screener/Android/app/src/main/java/ai/zetic/aiberry/emotion/EmotionStyle.kt
package ai.zetic.aiberry.emotion

import androidx.compose.ui.graphics.Color
import ai.zetic.aiberry.ui.Theme

/** Visual identity (emoji + color) for each emotion label, for a polished result UI. */
object EmotionStyle {
    fun emoji(label: String): String = when (label) {
        "Angry" -> "😠"      // 😠
        "Disgust" -> "🤢"    // 🤢
        "Fear" -> "😨"       // 😨
        "Happy" -> "😄"      // 😄
        "Neutral" -> "😐"    // 😐
        "Sad" -> "😢"        // 😢
        "Surprise" -> "😲"   // 😲
        else -> "🎙️"   // 🎙️
    }

    fun color(label: String): Color = when (label) {
        "Angry" -> Color(0.90f, 0.30f, 0.30f)
        "Disgust" -> Color(0.45f, 0.65f, 0.30f)
        "Fear" -> Color(0.55f, 0.40f, 0.85f)
        "Happy" -> Color(0.97f, 0.74f, 0.20f)
        "Neutral" -> Color(0.55f, 0.58f, 0.62f)
        "Sad" -> Color(0.30f, 0.55f, 0.90f)
        "Surprise" -> Color(0.95f, 0.55f, 0.25f)
        else -> Theme.Accent
    }
}
